browser/components/shared/list/CheckBoxList.js

```diff
@@ -1,40 +1,81 @@
+import { useState, useEffect } from 'react';
 import styled from "styled-components";
 import PropTypes from 'prop-types';
 import Checkbox from '../inputs/CheckBox';
 
 function CheckBoxList(props) {
+    const [selectedItems, setSelectedItems] = useState([]);
+    const [listItems, setListItems] = useState([]);
+    var currentGroup = null;
+    function onChangeCheckbox(value) {
+        let index = selectedItems.indexOf(value);
+        let array = [...selectedItems];
+        if (index > -1) {
+            array.splice(index, 1);
+        } else {
+            array.push(value);
+        }
+        setSelectedItems(array);
+    }
+
+    const getSelectedItems = () => {
+        let returnData = [];
+        listItems.map((item) => {
```

Comment:
`Array.map( ... )` returns an array, so you can make it more terse and avoid the `returnData` array
```
return listItems.map((item) => {
    if (selectedItems.indexOf(item.value) > -1) {
        return item;
    }
});
```

Reply:
Good suggestion actually. Shouldn't we use `filter` though? Since map returns null for other records in this case?

Reply:
True, we should use filter instead as we don't don't modify the individual items. We just want a subset.
Good point! 💯